tests/unit/test_charm.py

```diff
@@ -132,19 +131,124 @@ def _set_other_node_api_address_in_peer_relation(self, relation_id: int, unit_na
             key_values=key_values,
         )
 
+    @patch("charm.generate_vault_unit_certificate")
+    @patch("ops.model.Container.push", new=Mock)
+    @patch("ops.model.Model.get_binding")
     @patch("ops.model.Container.remove_path")
-    def test_given_can_connect_to_workload_when_install_then_existing_data_is_removed(
-        self, patch_remove_path
+    def test_given_not_leader_when_install_then_existing_data_is_removed(
+        self, patch_remove_path, patch_get_binding, patch_generate_vault_unit_certificate
     ):
+        bind_address = "1.2.1.2"
+        ingress_address = "2.3.3.3"
         self.harness.set_leader(is_leader=False)
         self.harness.set_can_connect(container=self.container_name, val=True)
+        relation_id = self._set_peer_relation()
+        patch_get_binding.return_value = MockBinding(
+            bind_address=bind_address, ingress_address=ingress_address
+        )
+        patch_generate_vault_unit_certificate.return_value = (
+            "whatever certificate",
+            "whatever private key",
+        )
+        self._set_ca_certificate_secret_in_peer_relation(
+            relation_id=relation_id,
+            private_key="whatever private key",
+            certificate="whatever certificate",
+        )
 
         self.harness.charm.on.install.emit()
 
         patch_remove_path.assert_has_calls(
             calls=[call(path="/vault/raft/vault.db"), call(path="/vault/raft/raft/raft.db")]
         )
 
+    @patch("ops.model.Container.push")
+    @patch("ops.model.Model.get_binding")
+    @patch("charm.generate_vault_unit_certificate")
+    def test_given_non_leader_and_ca_cert_is_created_when_install_then_unit_cert_is_generated(
+        self,
+        patch_generate_unit_cert,
+        patch_get_binding,
+        patch_push,
+    ):
+        ca_certificate = "ca certificate"
+        ca_private_key = "ca private key"
+        unit_private_key = "unit private key"
+        unit_certificate = "unit certificate"
+        bind_address = "1.2.1.2"
+        ingress_address = "10.1.0.1"
```

Comment:
Those parameters could be defined as global as they are defined several times in the tests.

Reply:
I disagree, each of those is used in the scope of their own test. It's true that we ended up using the same values in different tests but they are not "global"